pkg/command/command.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package command
+
+import (
+	"os/exec"
+	"reflect"
+	"strings"
+)
+
+// ExecTag stands for the path of executable file in command.
+// If we want this util works ,
+// we must add Exec in the struct and use NewExec() to initialize it,
+// because the default way to initialize Exec means None in code.
+const ExecTag = "exec"
+
+// SubCommandTag stands for the sub command in common command.
+// We can use it in struct fields as a tag.
+// Just like MatchExtension below
+// type Iptables Struct {
+// 	Exec
+//	MatchExtension Match `sub_command:""`
+// }
+//
+// type Match Struct {
+// 	Exec
+//	Port string `para:"-p"`
+// }
+// Field with SubcommandTag needs to be a struct with Exec.
+const SubCommandTag = "sub_command"
+
+// ParaTag stands for parameters in command.
+// We can use it in struct fields as a tag.
+// Just like Port below
+// type Iptables Struct {
+// 	Exec
+//	Port string `para:"-p"`
+// }
+// If the field is not string type or []string type , it will be skipped.
+// If the tag value like "-p" is empty string ,
+// the para will just add the field value into the command just as some single value parameter in command.
+// If the value of field is empty string or empty string slice or empty slice, the field and tag will all be skipped.
+const ParaTag = "para"
+
+// Exec is the interface of a command.
+// We need to inherit it in the struct of command.
+// User must add ExecTag as the tag of Exec field.
+// Example:
+// type Iptables struct {
+//	Exec           `exec:"iptables"`
+//	Tables         string `para:"-t"`
+// }
+type Exec struct {
```

Comment:
It seems that the only function of `Exec` is to set `exec` tag? Then why it has an option field (and the field is verified to be not empty in `SearchKey`)?

Reply:
In the case of sub command , we need to know if the sub command is used.
Such as the Match subcommand in IptablesTest below.